core/src/main/java/com/opensymphony/xwork2/util/AnnotationUtils.java

```diff
@@ -151,4 +152,33 @@ public static String resolvePropertyName(Method method) {
 
         return ann;
     }
+
+    /**
+     * Returns the annotation on the given class or the package of the class. This searches up the
+     * class hierarchy and the package hierarchy for the closest match.
+     *
+     * @param <T>             class type
+     * @param clazz           The class to search for the annotation.
+     * @param annotationClass The Class of the annotation.
+     * @return The annotation or null.
+     */
+    public static <T extends Annotation> List<T> findAnnotations(Class<?> clazz, Class<T> annotationClass) {
+        List<T> anns = new ArrayList<>();
+
+        T ann = clazz.getAnnotation(annotationClass);
+        if (ann != null) {
+            anns.add(ann);
+        }
+
+        ann = clazz.getPackage().getAnnotation(annotationClass);
+        if (ann != null) {
+            anns.add(ann);
+        }
+
+        if (clazz.getSuperclass() != Object.class) {
+            anns.addAll(findAnnotations(clazz.getSuperclass(), annotationClass));
```

Comment:
if someone calls this method with Object.class, the recursive call ends with NPE

Reply:
I think that's good, calling this function in `Object` is a bad idea anyway.